src/algorithm.jl

```diff
@@ -59,6 +59,12 @@ type PODNonlinearModel <: MathProgBase.AbstractNonlinearModel
     # Domain Reduction
     bound_basic_propagation::Bool                               # Conduct basic bound propagation
 
+    # embedding formulation
+    embedding::Bool
+    embedding_encode::Any                                       # Encoding method used for embedding
+    embedding_ibs::Bool                                         # Enable independent branching scheme
+    embedding_link::Bool                                         # Linking constraints between x and α, type 1 usse hierarchical and type 2 with big-m
```

Comment:
What is this embedding_link for?

Reply:
This is for an experiment to try out the alternative bounding constraints (y1b0+y1b1<=x<=y1b1+y2b2) when you have log # of binary variables.

Reply:
Was this experiment successful - did it improve computation time and bounds?

Reply:
The short answer is: it can be good. The results are not consistent for two reasons: 1) in some cases, this can help, 2)the test scope is small. I suggest we keep the current code for later experiments.
For example, with the multiKND problem,
![image](https://user-images.githubusercontent.com/11194970/41553745-aa12bb20-7300-11e8-845a-213be721e4c2.png)
Another example with nlp3:
![image](https://user-images.githubusercontent.com/11194970/41553835-ee568de8-7300-11e8-8e74-84f5629e00fd.png)